core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1899,6 +1899,54 @@ object Stream {
     def foldMonoid(implicit O: Monoid[O]): Stream[F, O] =
       self.fold(O.empty)(O.combine)
 
+    def groupWithin(n: Int, d: FiniteDuration)(
+        implicit F: Effect[F],
+        timer: Timer[F],
+        ec: ExecutionContext): Stream[F, Segment[O, Unit]] = {
```

Comment:
In general given that `Segment`s can be potentially infinite while the `Segment`s generated by this can only be finite (which is very valuable for downstream consumers to know), it seems like you'd probably either want to restrict the type to a `Chunk` or potentially a non-`Segment` collection (e.g. `List`).

Reply:
I disagree here; I think it's fine to expose segment with a guarantee in the docs that these segments will be non-empty and finite.

Reply:
Eh... I don't think `Segment` buys you much at all here. Given most use cases I think of for this method, you end up working with the entire batch at once, i.e. you'll have some function of `Segment[O, Unit] => X` which almost certainly means you're going to call `.force.toList` or something similar on it immediately. I can only see `Segment` being useful if you `Pull.output` some a transformed version of it back into a new `Stream`, but if you're going to do that (e.g. with `map` or the like, you could probably have just done that on the parent `Stream` directly. If you just started out with a `List` (or `Vector`), you'll probably have better perf and save a conversion since far far more APIs are going to take a `List` than a `Segment`.